internal/clisurface/lint.go

```
@@ -0,0 +1,685 @@
+//
+// Fenced code blocks are parsed as shell: line continuations are joined,
+// pipelines are split, and only segments whose argv[0] is the CLI binary are
+// checked — every flag of every other tool in an example pipeline is ignored,
+// which is what keeps the false-positive rate at zero. Prose outside fences is
+// checked more loosely: only backticked long-flag tokens, and only against the
+// union of every flag in the tree, because prose rarely says which command it
+// means.
+//
+// The argv[0] rule is what buys the zero false-positive rate, and it costs
+// reach: an invocation the binary does not lead — "sudo brutus …", "time brutus
+// …", "FOO=bar brutus …" — is skipped rather than checked. Recognising prefixes
```

golangci-lint (2.12.2)

[error] `Recognising` is a misspelling of `Recognizing`

(misspell)